File: benchmarks/values/hash_freq.go
package main

import (
	"fmt"
	"strconv"
	"time"
)

func main() {
	t0 := time.Now()

	const numWords = 1000
	const numIter = 5000000

	words := make([]string, numWords)
	for i := 0; i < numWords; i++ {
		words[i] = "word" + strconv.Itoa(i)
	}

	freq := make(map[string]int, numWords)
	seed := uint32(42)
	for i := 0; i < numIter; i++ {
		seed = (seed*1103515245 + 12345) & 0x7FFFFFFF
		word := words[seed%numWords]
		freq[word]++
	}

	maxFreq := 0
	for _, v := range freq {
		if v > maxFreq {
			maxFreq = v
		}
	}

	elapsed := time.Since(t0)
	fmt.Println(maxFreq)
	fmt.Printf("elapsed: %.3fs\n", elapsed.Seconds())
}

File: benchmarks/values/hash_freq.py
import time

t0 = time.perf_counter()

num_words = 1000
num_iter = 5_000_000

words = [f"word{i}" for i in range(num_words)]

freq = {}
seed = 42
for _ in range(num_iter):
    seed = (seed * 1103515245 + 12345) & 0x7FFFFFFF
    word = words[seed % num_words]
    freq[word] = freq.get(word, 0) + 1

max_freq = max(freq.values())

t1 = time.perf_counter()
print(max_freq)
print(f"elapsed: {t1 - t0:.3f}s")
